### PR TITLE
feat(svm): add a flag field to projects to explicitly enable VM

### PR DESCRIPTION
**Pull request by alexbrdn**
- add a flag for enabling Security Vulnerability Monitoring to projects
- change display of vulnerabilities counts to have red background only when SVM is enabled
- fix generating and applying a moderation request with boolean and int32 fields

### CODE REVIEW

**Comment by mcjaeger** [comment]
works for editing, but I wanted also to test moderation. However, since the branch is older as the moderation branch with the requesting group, the listing of moderation requests fails. So I cannot check what happens when a non-allowed user sends a moderation request just changing the check box value. (exception below)

@alexbrdn could either please rebase the branch or test on your test system (if that is still the old version) if the moderation request things works?

```
org.ektorp.DbAccessException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "requestingUserGroup" (class org.eclipse.sw360.datahandler.thrift.moderation.ModerationRequest), not marked as ignorable (45 known properties: "licenseAdditionsIsSet", "requestDocumentDeleteIsSet", "reviewer", "documentName", "releaseAdditionsIsSet", "moderationState", "user", "projectDeletions", "moderationStateIsSet", "userIsSet" [truncated]])
 at [Source: org.ektorp.http.StdHttpResponse$ConnectionReleasingInputStream@1ef81220; line: 3, column: 4107] (through reference chain: org.eclipse.sw360.datahandler.thrift.moderation.ModerationRequest["requestingUserGroup"])
        at org.ektorp.util.Exceptions.propagate(Exceptions.java:19)
        at org.ektorp.http.RestTemplate.handleResponse(RestTemplate.java:128)
        at org.ektorp.http.RestTemplate.getUncached(RestTemplate.java:27)
        at org.ektorp.impl.DefaultQueryExecutor.executeQuery(DefaultQueryExecutor.java:50)
        at org.ektorp.impl.StdCouchDbConnector.executeQuery(StdCouchDbConnector.java:433)
        at org.ektorp.impl.StdCouchDbConnector.queryView(StdCouchDbConnector.java:429)
        at org.ektorp.support.CouchDbRepositorySupport.queryView(CouchDbRepositorySupport.java:210)
        at org.eclipse.sw360.moderation.db.ModerationRequestRepository.getRequestsByModerator(ModerationRequestRepository.java:65)
```

**Comment by mcjaeger** [comment]
Hm, I think I am blocked as I cannot open projects anymore, I tried now testing #542 , but I got the error that "projects is not available" (see exception below) I am puzzeled, because I tried 
* delete sw360 artifacts from the maven local repo
* complete undeploy of all sw360 wars and redeployed the wars from #542 
I think it would be good to merge this branch quickly and rebase project referring branches then ( @imaykay ). I you have other ideas please let me know.

**Comment by alexbrdn** [comment]
@mcjaeger , looks like moderation requests in your database have been migrated by script 007 and have the new field introduced by #529. To test this branch, you have to roll back the database, or wait until #529 is merged and this PR is rebased

**Comment by imaykay** [comment]
I have tried to run the branch beside the code review and always get `2017-08-25T14:08:46.160252371Z 14:08:45,154 ERROR [http-apr-8080-exec-21][render_portlet_jsp:132] null`
`2017-08-25T14:08:46.160256866Z javax.el.PropertyNotFoundException: Property 'enableSvm' not found on type org.eclipse.sw360.datahandler.thrift.projects.Project`
`2017-08-25T14:08:46.160261560Z 	at javax.el.BeanELResolver$BeanProperties.get(BeanELResolver.java:268)
`
even though I see the property in the generated source code and the deployment seems to run. Container and Docker restart with redeployments does not solve the issue. I have to recheck on monday what is happening here...

**Comment by imaykay** [comment]
Okay, after having spent some time on this it seems like the deployment via chores is not yet optimal as old files are not deleted.
doubleddependencies

So after tagging a release and increasing the version in commit https://github.com/sw360/sw360portal/commit/368caa5ad892e08b51e6f3790be5fff324678a21 users of chores were not able to see changes done in the 2.3.0 artefacts of dependencies, especially in thrift services.

To fix that, one needs to navigate to 
`cd sw360-sw360chores/deployment`

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf [hunk 1]
```diff
@@ -97,6 +97,19 @@
                       placeholder="Enter Description"><sw360:out value="${project.description}" stripNewlines="false"/></textarea>
         </td>
     </tr>
+    <tr>
+        <td width="33%">
+            <label class="checkboxlabel" for="enable_svm">
+                <input id="enable_svm" type="checkbox"
+                       name="<portlet:namespace/><%=Project._Fields.ENABLE_SVM%>"
+                       value="<sw360:out value="${project.enableSvm}"/>"
+                        <core_rt:if test="${project.enableSvm == 'true'}"> checked="checked" </core_rt:if>
```
Comment (imaykay) [inline frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf]: You could skip the `== 'true'` part in my opinion.

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/vulnerabilities.jspf [hunk 2]
```diff
@@ -25,7 +25,12 @@
 <jsp:useBean id="vulnerabilityCheckstatusTooltips" type="java.util.Map<java.lang.String, java.lang.String>" scope="request"/>
 <jsp:useBean id="vulnerabilityMatchedByHistogram" type="java.util.Map<java.lang.String, java.lang.Integer>" scope="request"/>
 
-
+<p>Security Vulnerability Monitoring:
+        <span
+            <core_rt:if test="${project.enableSvm == 'true'}">class="notificationBulletInline backgroundOK">enabled</core_rt:if>
+            <core_rt:if test="${project.enableSvm != 'true'}">class="notificationBulletInline backgroundGrey">disabled</core_rt:if>
```
Comment (imaykay) [inline frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/vulnerabilities.jspf]: Same here, maybe just `${not project.enableSvm}`? Also in line above this one.

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf (hunk 1 above)
Comment (alexbrdn) [reply frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf]: You're right. That's a waste of some characters. Now corrected

**Comment by alexbrdn** [comment]
@imaykay, please review again

**Pull request merged**